imperial_coldfront_plugin/views.py

```diff
@@ -43,6 +44,23 @@
 User = get_user_model()
 
 
+class GraphAPISearch(UserSearch):
+    """Search for users using MS Graph API."""
+
+    def search_a_user(
+        self, user_search_string: str | None = None, search_by: str = "all_fields"
+    ) -> list[str]:
+        """Searchers for a single user.
+
+        Args:
+            user_search_string: The user string to look for.
+            search_by: (Unused) Fields to look into. This backend always looks into the
+                user's name or username.
+        """
+        graph_client = get_graph_api_client()
+        return graph_client.user_search(user_search_string)
```

Comment:
Data returned here isn't in the expected format for the search backend interface. You can see the expected format [here](https://github.com/ubccr/coldfront/blob/0b96bd391136d0a62ee1840e3cae111e6904b999/coldfront/core/user/utils.py#L61-L67). It's pretty similar but we should add the missing keys. You may need to pull extra data back from the graph api for the first/last names. I'm not sure how reliable splitting the full name would be.